hathor/verification/block_verifier.py

```diff
@@ -42,9 +42,35 @@ def verify(self, block: Block) -> None:
             # TODO do genesis validation
             return
 
-        block.verify_without_storage()
+        self.verify_without_storage(block)
 
         # (1) and (4)
-        block.verify_parents()
+        self.verify_parents(block)
+
+        self.verify_height(block)
+
+    def verify_without_storage(self, block: Block) -> None:
+        """ Run all verifications that do not need a storage.
+        """
+        block.verify_without_storage()
 
+    def verify_height(self, block: Block) -> None:
+        """Validate that the block height is enough to confirm all transactions being confirmed."""
         block.verify_height()
+
+    def verify_weight(self, block: Block) -> None:
+        """Validate minimum block difficulty."""
+        block.verify_weight()
+
+    def verify_reward(self, block: Block) -> None:
+        """Validate reward amount."""
+        block.verify_reward()
+
+    def verify_no_inputs(self, block: Block) -> None:
+        block.verify_no_inputs()
+
+    def verify_outputs(self, block: BaseTransaction) -> None:
```

Comment:
typo `Block`?

Reply:
This is intended, as it overrides `VertexVerifier.verify_outputs()`, which has this signature. Analogously to https://github.com/HathorNetwork/hathor-core/pull/797#discussion_r1366090478, this will be improved in a new refactor after all these ones.

Reply:
Ok!